src/input.ts

```diff
@@ -7,14 +7,19 @@ export type ZipFileDescription = {
   bytes: ReadableStream<Uint8Array> | Uint8Array | Promise<Uint8Array>
   crc?: number // will be computed later
 }
+export type ZipFolderDescription = {
+  modDate: Date
+  bytes?: undefined
+  crc?: undefined
+}
 
 /** The file name and modification date will be read from the input if it is a File or Response;
  * extra arguments can be given to override the input's metadata.
  * For other types of input, the `name` is required and `modDate` will default to *now*.
  * @param encodedName will be coerced to string, so… whatever
  * @param modDate should be a Date or timestamp or anything else that works in `new Date()`
  */
-export function normalizeInput(input: File | Response | BufferLike | StreamLike, modDate?: any): ZipFileDescription {
+export function normalizeInput(input?: File | Response | BufferLike | StreamLike, modDate?: any): ZipFileDescription | ZipFolderDescription {
```

Comment:
It would be more accurate to type the relationship between the `input` parameter and output, instead of widening types on both ends. Either using a ternary or an overload (I prefer the overload, so you can have a different JSDoc to explain what's happening).

Otherwise, there's no point having two separate types.

Reply:
Good idea. Done.